### PR TITLE
Switch to using the DAGMC targets in CMake.

### PR DESCRIPTION
**Pull request by pshriwise**
DAGMC now exports targets on install (https://github.com/svalinn/DAGMC/pull/662).

This updates OpenMC to use those targets and simplify our CMake file a bit.

### CODE REVIEW

**Comment by paulromano** [comment]
Also, thanks for cleaning this up!

**Comment by pshriwise** [comment]
Ah yeah, thanks for catching the `OpenMCConfig.cmake.in` removals. They're gone now.

As for the `FindDAGMC.cmake` it seems like we can indeed get rid of it as long as the `CMAKE_PREFIX_PATH` is set correctly.

File: CMakeLists.txt [hunk 1]
```diff
@@ -371,7 +364,7 @@ target_link_libraries(libopenmc ${ldflags} ${HDF5_LIBRARIES} ${HDF5_HL_LIBRARIES
 
 if(dagmc)
   target_compile_definitions(libopenmc PRIVATE DAGMC)
-  target_link_libraries(libopenmc dagmc-imported)
+  target_link_libraries(libopenmc dagmc-shared pyne_dagmc-shared uwuw-shared)
```
Comment (paulromano) [inline CMakeLists.txt:367]: Why does `pyne_dagmc-shared` need to be included?

File: tools/ci/travis-install.py [hunk 2]
```diff
@@ -47,6 +47,8 @@ def install(omp=False, mpi=False, phdf5=False, dagmc=False):
 
     if dagmc:
         cmake_cmd.append('-Ddagmc=ON')
+        home_dir = os.environ.get('HOME')
+        cmake_cmd.append('-DCMAKE_PREFIX_PATH={}/DAGMC/lib'.format(home_dir))
```
Comment (paulromano) [inline tools/ci/travis-install.py:51]:
```suggestion
```

File: CMakeLists.txt [hunk 3]
```diff
@@ -40,13 +40,6 @@ endif()
 #===============================================================================
 if(dagmc)
   find_package(DAGMC REQUIRED)
```
Comment (paulromano) [inline CMakeLists.txt:42]:
```suggestion
  find_package(DAGMC REQUIRED PATH_SUFFIXES lib/cmake)
```

File: CMakeLists.txt (hunk 1 above)
Comment (pshriwise) [reply CMakeLists.txt:367]: It seems it isn't. I'd mistakenly thought it was necessary due to the references to all of the PyNE objects in the UWUW library. Thanks for catching that!

**Pull request merged**